libensemble/alloc_funcs/start_persistent_consensus.py

```diff
@@ -0,0 +1,316 @@
+    consensus, which is just a linear combination of a nodes's neighbors'
+    solutions. Thus, we develop a general alloc function that does both of
+    these. 
+
+    From the caller function, the user must pass a square matrix in
+    `persis_info['A']. Typically, this is a Laplacian matrix of a connected
+    graph or a doubly stochastic matrix. Note that the alloc will send the
+    row i's non-zero indices values to gen i when initializing the gen
+    via the persis_info object. This information is used methods such as
+    `gen_funcs/persistent_n_agent.py`, where each gen needs to take
+    a linear combination of its neighbors's `x`'s. 
+
+    The user has three remaining optional values to pass into the alloc,
+    which are all set from the calling script. 
+
+    - 1. The user can set `persis_info['print_progress']=1` 
+    This tells alloc function print the consensus value and iteration count
+    whenever a consensus step is taken
+
+    - 2. The user can set `persis_info['gen_params']={... dictionary ...}`. 
+    The dictionary contains the any parameters that the gen will utilize.
+    This can include, for example, functions that the gen can use to compute
+    gradients rather than requesting a sim to complete the work (see the
+    linear regression tests in `tests/regression_tests/test_persistent_pds.py`)
+
+    - 3. The user can set `persis_info['sim_params']={... dictionary ...}`. 
+    Similarly, the dictionary contains the any parameters that the sim will
+    utilize   
+
+                ------------------------------------------------
+
+    Now, we briefly explain how to request the gradient of the consensus term
+    or the gradient of the function as the gen. 
+
+    For the former, the gen submits a work request and must set the parameter 
+    `consensus_pt` to True, while making sure to pass into the `x` variable. 
+
+    If the user wants the alloc to sum all the {f_i}, the user must set both 
+    `consensus_pt` and `eval_pt` to True while settings the `f_i` variable. (The 
+    reason for having both variables set to True is to simplify the implemenation 
+    in the alloc.)
+
+    Finally, to request a gradient of f_i, `consensus_pt` must be set to False,
+    `get_grad` needs to be set to True, `obj_component` (i.e., which f_i to 
+    consider) must be set, and the `x` variable must be set.  
+
+    If the user wants a function evaluation, then set `get_grad` to False instead. 
+    """
+
+    Work = {}
+    gen_count = count_persis_gens(W)
+    is_first_iter = False
+
+    if persis_info.get('first_call', True):
+        persis_info.update({'last_H_len': 0})
+        persis_info.update({'next_to_give': 0})
+        persis_info.update({'first_call': False})
+        is_first_iter = True
+
+        A = persis_info.get('A')
+        assert A.shape[0]==A.shape[1], 'Matrix @A is not square'
+        # assert la.norm(A.dot(np.ones(A.shape[1])))/la.norm(A.toarray()) < 1e-14, 'Matrix @A is not row stochastic'
```

Comment:
Should this assert be removed or uncommented?

Reply:
Remove